lib/admin/WP_Auth0_Admin_Advanced.php

```diff
@@ -134,294 +134,233 @@ public function render_passwordless_method() {
-?>
-      <input type="text" name="<?php echo $this->options->get_options_name(); ?>[cdn_url]" id="wpa0_cdn_url" value="<?php echo esc_attr( $cdn_url ); ?>"/>
-
-      <div class="subelement">
-        <span class="description"><?php echo __( 'Point this to the latest widget available in the CDN', 'wp-auth0' ); ?></span>
-      </div>
-    <?php
+  public function render_social_facebook_secret( $args ) {
+    $this->render_social_key_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Used for the Social Amplification Widget. ', 'wp-auth0' ) .
+      $this->get_docs_link(
+        'connections/social/facebook#5-get-your-app-id-and-app-secret',
+        __( 'Instructions here', 'wp-auth0' )
+      )
+    );
   }
 
-  public function render_auth0_server_domain() {
-    $v = $this->options->get( 'auth0_server_domain' );
-?>
-      <input type="text" name="<?php echo $this->options->get_options_name(); ?>[auth0_server_domain]" id="wpa0_auth0_server_domain" value="<?php echo esc_attr( $v ); ?>" />
+  public function render_valid_proxy_ip( $args ) {
+    $this->render_text_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Whitelist for proxy and load balancer IPs to enable logins and migration webservices', 'wp-auth0' )
+    );
+  }
 
-      <div class="subelement">
-        <span class="description"><?php echo __( 'The Auth0 domain, it is used by the setup wizard to fetch your account information.', 'wp-auth0' ); ?></span>
-      </div>
-    <?php
+  public function render_cdn_url( $args ) {
+    $this->render_text_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'This should point to the latest widget JS available in the CDN and rarely needs to change', 'wp-auth0' )
+    );
   }
 
-  public function render_connections() {
-    $v = $this->options->get( 'lock_connections' );
-?>
-      <input type="text" name="<?php echo $this->options->get_options_name(); ?>[lock_connections]" id="wpa0_connections" value="<?php echo esc_attr( $v ); ?>" />
+  public function render_auth0_server_domain( $args ) {
+    $this->render_text_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'The Auth0 domain used by the setup wizard to fetch your account information', 'wp-auth0' )
+    );
+  }
 
-      <div class="subelement">
-        <span class="description"><?php echo __( 'This is used to select which connections should lock show. It is ignored when empty and is mandatory for passwordless with social mode.', 'wp-auth0' ); ?></span>
-      </div>
-    <?php
+  public function render_connections( $args ) {
+    $this->render_text_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Connections the Auth0 login form should show (separate multiple with commas). ', 'wp-auth0' ) .
```

Comment:
Connections _that_ the .. 

and then

separate multiple connections with commas

(Same as in the IP text)